contracts/mocks/generic/CoverProductsGeneric.sol

```diff
@@ -6,6 +6,8 @@ import "../../interfaces/ICoverProducts.sol";
 
 contract CoverProductsGeneric is ICoverProducts {
```

Comment:
(No action required) This is just an observation and is out of scope for this PR.

The `CoverProductsGeneric` contract lacks any meaningful implementation, and its child contracts independently implement their own methods, leading to duplicate code.

In the future, we should follow the pattern of consolidating all shared logic into the Generic contract, leaving only specific use-case implementations in the child contracts.